chore: reorganize directory hierarchy

Patrik-Stas opened the pull request:
Further implementing https://github.com/hyperledger/aries-vcx/issues/1045

mirgee commented:
I am not sure about the grouping. Why should `public_key` belong under `did`? Don't `did_resolver_registry` and `did_resolver` more relate to `did_document` than `did`? Perhaps `did_document` might belong under `did?` The `tools` directory is confusing - can't crates like `public_key` and `did_resolver_registry` (and others) be seen as tools as well?

Patrik-Stas commented:
@mirgee Done some changes, let me know what you think. 

Changes:
- Renamed `tools` to `misc`
- Moved `public_key` to `/misc`
- Moved `did_resolver_registry` to `/aries/misc` as that's a component only used within `/aries` scope
- Merged `/did` and `/did_document` directories into `did_core`, as per name of the spec - hence removing dilemmas whether something belongs to `did` or `did_document`. DID Core spec talks about both, so I figured we could have it under single `/did_core` scope.

mirgee commented:
@Patrik-Stas I think this is much better :)
